### ISSUE TITLE
Button Link text color change when visited

### ISSUE THREAD

**Issue by grneggandsam** [cerner/terra-core#1212]
# Issue Description
The text of terra-button components changes to a blue color if it links to another page after the page has been visited.
![screen shot 2018-02-06 at 10 32 25 am](https://user-images.githubusercontent.com/12060103/35871126-16c7734a-0b29-11e8-8e3a-00435feb8163.png)

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior

The Button should continually display with the same color text (white in my case)

## Current Behavior

Buttons are now displaying with blue text after their link has been visited

## Steps to Reproduce

1. Create a terra button component with an href 
2. Visit the link
3. Go back to original page to see new text color

## Environment

* Component Version: Not completely sure, but I believe its the Terra-Base (version 2.11.0)
* Browser Name and Version: Chrome Version 63.0.3239.132 & Firefox 58.0.1 (64-bit)
* Operating System and version (desktop or mobile): macOS Sierra Version 10.12.6 (desktop)

**Comment by kafkahw** [comment]
Any updates on this? This is a big impact on our solution.

**Comment by bjankord** [comment]
We've identified the root cause on this. When we added global styles to all [hyperlinks to update them to the OCS colors](https://github.com/cerner/terra-core/pull/1171), we didn't account for needing to updated styles for the link pseudo-classes for our components that render as anchors. We've logged #1215 to dig into the underlying issue, it looks like terra-button is the only terra-core component to get hit by this bug.

We added these styles in terra-base v2.11.0.

To fix the issue in our code, we'll be adding a selector in terra-button to account for there now being global styles applied to `a:visited` to ensure those styles don't impact the terra-button when it is rendered as an anchor and a user has visited the anchor's href URL before.

With the current state of button in terra-core we are planning to release terra-button v2 shortly. This would mean to get the fix for this as a consumer, your apps would need to update to terra-button v2. As this may be a bigger ask than anticipated to get the bug fix, I am considering the steps we'd need to take to do a back patch release of terra-button v1.14.1 to resolve this in the v1 version of terra-button.

To help test against this issue with global link styles in the future, we'll be adding screenshot tests for button's rendered as anchors in all of the various pseudo-class states (:focus, :hover, :visited, :active) similar to how [we are capturing screenshots for hyperlinks rendered in all of these states](https://github.com/cerner/terra-core/blob/master/packages/terra-base/tests/nightwatch/Anchors.jsx).
If we add other components where we control the rendering and are rendering anchors, we'll add screenshot testing to test all the pseudo-class states as well

**Comment by bjankord** [comment]
This bug exists currently in our code for button v2 and will need to be fixed before we release terra-button@2.0.0

**Comment by kafkahw** [comment]
@bjankord Thanks for the update. Just FYI, this seems to be affecting other anchor links in our app as well besides button link. See image below.
The profile links on the bottom left are supposed to be black, but they are now blue because of visited styles. I feel that's something your team might want to take into account as well. I understand that we can change the visited color by setting css variable `--terra-link-visted-color`, but that will take a global effect and impact all anchor links. We might want different link colors in different area of our app.
<img width="884" alt="screen shot 2018-02-08 at 9 29 40 am" src="https://user-images.githubusercontent.com/2274728/35981410-cffd53b8-0cb2-11e8-8ba5-9f6923c6b476.png">

**Comment by alex-bezek** [comment]
also, thanks to @kafkahw noting that css variable, isn't it mispelled?
https://github.com/cerner/terra-core/blob/3f58fa9fff3f0be26305534b03ca54a11aa2ad4a/packages/terra-base/src/Base.scss#L45
```
a.is-visited {
  color: var(--terra-link-visted-color, #0065a3);
}
```
`visited` vs
`visted`

**Comment by bjankord** [comment]
These has been resolved in terra-base@2.11.1.
We've reverted the global link styles in the path release. We will be creating an anchor component in the future to provide a way to style links as the UX team intends.
terra-base@2.11.1 is now released and available on npm. cc/ @grneggandsam  @kafkahw
